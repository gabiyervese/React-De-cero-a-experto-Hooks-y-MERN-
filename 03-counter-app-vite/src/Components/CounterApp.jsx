import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value, name, luckyNumber }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleSubs = () => {
    setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>Counter App</h1>
      <h2 data-testid="test-title">{name}</h2>
      <h3>{counter}</h3>
      <p>{luckyNumber}</p>
      {/* <p>{value}</p> */}
      <button onClick={handleAdd}>+</button>
      <button onClick={handleSubs}>-</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number,
  name: PropTypes.string.isRequired,
};

CounterApp.defaultProps = {
  // name: "Gabriela",
  luckyNumber: "6",
};
