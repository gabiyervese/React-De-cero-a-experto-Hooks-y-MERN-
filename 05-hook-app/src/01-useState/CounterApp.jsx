import { useState } from "react";

const CounterApp = () => {
  const [counter, setCounter] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
  });

  const { counter1, counter2, counter3 } = counter;

  return (
    <>
      <h1>Lets count</h1>
      <h2>Counter1: {counter1}</h2>
      <h2>Counter2: {counter2}</h2>
      <h2>Counter3: {counter3}</h2>
      <button
        className="btn"
        onClick={() => setCounter({ ...counter, counter1: counter1 + 1 })}
      >
        + 1
      </button>
    </>
  );
};

export default CounterApp;
