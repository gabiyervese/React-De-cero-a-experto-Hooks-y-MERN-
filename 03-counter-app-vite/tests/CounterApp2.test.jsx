const { render, screen } = require("@testing-library/react"); //screen es el objeto que estamos renderizando.
import { CounterApp } from "../src/Components/CounterApp";

//ESTAS PRUEBAS SON MAS LIMPIAS QUE EL COUNTERAPPTEST ANTERIOR.

describe("Pruebas en <CounterApp />", () => {
  const name = "Mariana";
  const value = 18;
  test("debe hacer match con el snapshot", () => {
    const { container } = render(<CounterApp name={name} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar el nombre Mariana", () => {
    render(<CounterApp name={name} />);
    expect(screen.getByText(name)).toBeTruthy();
    //screen.debug() me muestra el objeto en consola.
  });

  test("Debe de mostrar el nombre en un h2", () => {
    render(<CounterApp name={name} />);
    expect(screen.getByRole("heading", { level: 2 })).toBeTruthy();
  });

  test("debe enviar el value enviado por props", () => {
    render(<CounterApp name={name} value={value} />);

    expect(screen.getAllByText(value).length).toBe(2);
  });
});
