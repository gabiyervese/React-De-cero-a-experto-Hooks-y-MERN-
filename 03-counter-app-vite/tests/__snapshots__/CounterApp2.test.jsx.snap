// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2
    data-testid="test-title"
  >
    Mariana
  </h2>
  <h3 />
  <p>
    6
  </p>
  <p />
  <button>
    +
  </button>
  <button>
    -
  </button>
  <button>
    Reset
  </button>
</div>
`;
